zookeeper-server/src/test/java/org/apache/zookeeper/test/EmptiedSnapshotRecoveryTest.java

```diff
@@ -119,7 +125,15 @@ public void testRestoreWithEmptySnapFiles() throws Exception {
      */
     @Test
     public void testRestoreWithNoSnapFiles() throws Exception {
-        runTest(false);
+        runTest(false, false);
+    }
+
+    @Test
+    public void testRestoreWithTrustedEmptySnapFiles() throws Exception {
+        FileTxnSnapLog.setTrustEmptySnapshotFlag(true);
```

Comment:
what about resetting the value to "false" in a finally block, this way if the test fails we will be sure that the rest of the tests are executed with the default value

Reply:
good catch, and find bug complains about something similar. I changed the flag variable to an instance variable instead and move the set / check logic inside the shared test implementation. Tests will not be interfering with each other.